PORT-5679 Overhaul GitLab installation page (#756)
port-labs/port-docs: docs/build-your-software-catalog/sync-data-to-catalog/git/gitlab/installation.md

- How to [deploy](#deploying-the-gitlab-integration) the integration in the configuration that fits your use case.

:::note Prerequisites

- A gitlab account with admin privileges.
- A gitlab group account with the `api` scope.
- If you choose the real time & always on installation method - a kubernetes cluster to install the integration on.
- Your Port user role is set to `Admin`.

:::

## Creating a GitLab group access token

A group access token can be used for the group it was generated at, as well as for all sub-groups underneath it.

The GitLab integration is able to query multiple GitLab root groups. To do so, it will require multiple group access tokens, each at the correct root group.

<details>
<summary>GitLab group access tokens example</summary>

For example, let's assume the following GitLab account structure:

```
GitLab account
.
├── microservices-group
│   ├──microservice1-group
│   └──microservice2-group
├── apis-group
│   ├── rest-api-group
│   └── graphql-api-group
```

In this example:

- To map the `microservices-group`, we require only one group access token - one for the `microservices-group`.

> - To map the `microservices-group` **and** all of its subgroups, we require only one group access token - one for the `microservices-group`.

Reviewer: This section is repetitive and a bit messy, I would simplify it.
Also, I think it should be worded like this:
To map the `microservices-group` **and** all of its subgroups, we require only one group access token.

- To map the `apis-group`, we require only one group access token - one for the `apis-group`.
- To map the `apis-group` **and** all of its subgroups, we require only one group access token - one for the `apis-group`.
- To map the `microservices-group`, **the** `apis-group` **and** all of their subgroups, we require only two group access tokens - one for the `microservices-group` and one for the `apis-group`.
- To map the `microservice1-group`, we have 2 options:
  - Create a group access token for the `microservices-group` and use the [token mapping](#tokenmapping) to select just the `microservice1-group`.
  - Create a group access token for the `microservice1-group` directly.

</details>

See the [token mapping](#tokenmapping) section for more information.

The following steps will guide you how to create a GitLab group access token.

1. Sign in to GitLab and go to your desired group's settings page:

>    ![GitLab group settings](../../../../../static/img/integrations/gitlab/GitLabGroupSettings.png)

Reviewer: Full path links plz 😸

Mapping format:

```text showLineNumbers
{"MY_FIRST_GITLAB_PROJECT_GROUP_TOKEN": ["*/MyFirstGitLabProject/**","*/MySecondGitLabProject/*"]}
```

Example:

```text showLineNumbers
{"glpat-QXbeg-Ev9xtu5_5FsaAQ": ["*/DevopsTeam/*Service", "*/RnDTeam/*Service"]}
```

Multiple GitLab group access tokens example:

```text showLineNumbers
{"glpat-QXbeg-Ev9xtu5_5FsaAQ": ["*/DevopsTeam/*Service", "*/RnDTeam/*Service"],"glpat-xF7Ae-vXu5ts5_QbEgAQ9": ["*/MarketingTeam/*Service"]}
```

### `appHost` & listening to hooks

:::tip
The `appHost` parameter is used specifically to enable the real-time functionality of the integration.

> If it is not provided, the integration will continue to function correctly, but in order to retrieve the latest information from the target system, it will require either an [event listener](https://ocean.getport.io/framework/features/event-listener) of type `SCHEDULE` or a manual resync periodically through Port's UI.

Reviewer 2: What type is schedule?

    This means that if you choose system hooks, the integration will not be able to update the relevant entities in Port on events such as `issues` or `pipeline`.

- Creating a system hook requires admin privileges in GitLab. Due to this, the integration supports that the system hook will be created manually, and the integration will use it to listen to the relevant events.

##### Configuring the integration to use hooks

By default, if `appHost` is provided, the integration will create group webhooks for each group in your GitLab instance.

To create a system hook there are two options:

:::note
In both options you'll need to provide the `useSystemHook` parameter with the value `true`.
:::

1. Provide a token with admin privileges in GitLab using the `tokenMapping` parameter.

>    1. In this option, the integration will create the system hook in your GitLab account automatically.

Reviewer: ```suggestion
   - When choosing this option, the integration will create the system hook in your GitLab account automatically.
```

2. Create the system hook manually

>    1. Follow the instructions for creating a system hook in GitLab [here](https://docs.gitlab.com/ee/administration/system_hooks.html#create-a-system-hook).

Reviewer: ```suggestion
   - Follow the instructions for creating a system hook in GitLab [here](https://docs.gitlab.com/ee/administration/system_hooks.html#create-a-system-hook).
```

>    2. In the `URL` field, provide the `appHost` parameter value with the path `/integration/system/hook`. e.g. `https://my-gitlab-integration.com/integration/system/hook`.

Reviewer: ```suggestion
   - In the `URL` field, provide the `appHost` parameter value with the path `/integration/system/hook`. e.g. `https://my-gitlab-integration.com/integration/system/hook`.
```

>    3. From the `Triggers` section, the GitLab integration currently supports the following events:

Reviewer: ```suggestion
   - From the `Triggers` section, the GitLab integration currently supports the following events:
```

>       1. `push`

Reviewer: ```suggestion
      - `push`
```

>       2. `merge_request`

Reviewer: ```suggestion
      - `merge_request`
```

:::

Make sure to configure the following [Github Secrets](https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions):

<DockerParameters/>

<br/>

Here is an example for `gitlab-integration.yml` workflow file:

```yaml showLineNumbers
name: GitLab Exporter Workflow

# This workflow responsible for running GitLab exporter.

on:
  workflow_dispatch:

jobs:
  run-integration:
    runs-on: ubuntu-latest

    steps:
      - name: Run GitLab Integration
        run: |
          # Set Docker image and run the container
          integration_type="gitlab"
          version="latest"

          image_name="ghcr.io/port-labs/port-ocean-$integration_type:$version"

          docker run -i --rm --platform=linux/amd64 \
          -e OCEAN__EVENT_LISTENER='{"type":"ONCE"}' \
          -e OCEAN__INITIALIZE_PORT_RESOURCES=true \
          -e OCEAN__INTEGRATION__CONFIG__TOKEN_MAPPING="${{ secrets.OCEAN__INTEGRATION__CONFIG__TOKEN_MAPPING }}" \
          -e OCEAN__PORT__CLIENT_ID=${{ secrets.OCEAN__PORT__CLIENT_ID }} \
          -e OCEAN__PORT__CLIENT_SECRET=${{ secrets.OCEAN__PORT__CLIENT_SECRET }} \
          $image_name

          exit $?
```

  </TabItem>

>   <TabItem value="jenkins" label="Jenkins">

Reviewer: ```suggestion
  <TabItem value="jenkins" label="Jenkins">
  
```

Author: What is the exact change here? the lines look the same

Reviewer: just a line break between <TabItem> and the next line:)
